src/ripple/app/main/Application.cpp

```diff
@@ -1978,49 +2079,31 @@ ApplicationImp::journal (std::string const& name)
     return logs_->journal (name);
 }
 
-//VFALCO TODO clean this up since it is just a file holding a single member function definition
-
-static
-std::vector<std::string>
-getSchema (DatabaseCon& dbc, std::string const& dbName)
+void
+ApplicationImp::updateTxnDB()
 {
-    std::vector<std::string> schema;
-    schema.reserve(32);
-
-    std::string sql = "SELECT sql FROM sqlite_master WHERE tbl_name='";
-    sql += dbName;
-    sql += "';";
-
-    std::string r;
-    soci::statement st = (dbc.getSession ().prepare << sql,
-                          soci::into(r));
-    st.execute ();
-    while (st.fetch ())
+    auto schemaHas = [&](std::string const& column)
     {
-        schema.emplace_back (r);
-    }
-
-    return schema;
-}
+        std::string cid, name;
+        soci::statement st = (mTxnDB->getSession().prepare <<
+            ("PRAGMA table_info(AccountTransactions);"),
+            soci::into(cid),
+            soci::into(name));
 
-static bool schemaHas (
-    DatabaseCon& dbc, std::string const& dbName, int line,
-    std::string const& content, beast::Journal j)
-{
-    std::vector<std::string> schema = getSchema (dbc, dbName);
+        st.execute();
+        while (st.fetch())
+        {
+            if (name == column)
+                return true;
+        }
 
-    if (static_cast<int> (schema.size ()) <= line)
-    {
-        JLOG (j.fatal()) << "Schema for " << dbName << " has too few lines";
-        Throw<std::runtime_error> ("bad schema");
-    }
+        return false;
+    };
 
-    return schema[line].find (content) != std::string::npos;
-}
+    assert(schemaHas("TransID"));
+    assert(!schemaHas("foobar"));
```

Comment:
Did you want to keep the `foobar` assert?

Reply:
I don't know the reason it was previously added, so I am opting to keep it.